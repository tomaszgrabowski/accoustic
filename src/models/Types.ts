export type HeaderSize = 1 | 2 | 3 | 4 | 5 | 6;
export type HeaderType = 'article-header' | 'text-center';
