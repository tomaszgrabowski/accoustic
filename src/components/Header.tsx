import React from 'react';
import Navigation from './Navigation';

interface IProps {
    appName: string;
}

const Header = ( props: IProps ) => {
    return (
        <header>
            <h1>{ props.appName }</h1>
            <Navigation/>
        </header>
    );
};

export default Header;
