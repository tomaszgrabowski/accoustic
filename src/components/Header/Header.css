.text-center {
    text-align: center;
}

.article-header {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 5rem;
    background-color: silver;
}
