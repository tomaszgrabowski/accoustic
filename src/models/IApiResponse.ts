export interface IApiResponse {
    rev: string;
    thumbnail: Thumbnail;
    keywords: any[];
    kind: any[];
    created: string;
    creatorId: string;
    description: string;
    classification: string;
    type: string;
    locale: string;
    tags: any[];
    selectedLayouts: SelectedLayout[];
    elements: Elements;
    name: string;
    lastModifierId: string;
    typeId: string;
    links: Links;
    id: string;
    lastModified: string;
    systemModified: string;
    status: string;
}

interface Links {
    thumbnail: Thumbnail2;
    retire: Thumbnail2;
    draft: Thumbnail2;
    self: Thumbnail2;
    type: Thumbnail2;
}

interface Thumbnail2 {
    href: string;
}

interface Elements {
    heading: Heading;
    author: Heading;
    body: Body;
    date: Heading;
    mainImage: MainImage;
}

interface MainImage {
    elementType: string;
    value: Value;
    typeRef: Layout;
}

interface Value {
    leadImage: LeadImage;
    leadImageCaption: Heading;
    leadImageCredit: Heading;
}

interface LeadImage {
    mode: string;
    profiles: string[];
    renditions: Renditions;
    asset: Asset;
    elementType: string;
    url: string;
}

interface Asset {
    fileName: string;
    altText: string;
    fileSize: number;
    width: number;
    mediaType: string;
    id: string;
    resourceUri: string;
    height: number;
}

interface Renditions {
    lead: Lead;
    card: Lead;
    default: Default;
}

interface Default {
    width: number;
    source: string;
    height: number;
    url: string;
}

interface Lead {
    source: string;
    width: number;
    height: number;
    transform: Transform;
    url: string;
}

interface Transform {
    scale: number;
    crop: Crop;
}

interface Crop {
    x: number;
    y: number;
    width: number;
    height: number;
}

interface Body {
    values: string[];
    elementType: string;
}

interface Heading {
    elementType: string;
    value: string;
}

interface SelectedLayout {
    layout: Layout;
}

interface Layout {
    id: string;
}

interface Thumbnail {
    id: string;
    url: string;
}
