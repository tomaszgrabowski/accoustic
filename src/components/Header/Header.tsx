import React from 'react';
import Navigation from '../Navigation/Navigation';
import './Header.css';

interface IProps {
    appName: string;
}

const Header = ( props: IProps ) => {
    return (
        <header>
            <h1>{ props.appName }</h1>
            <Navigation/>
        </header>
    );
};

export default Header;
