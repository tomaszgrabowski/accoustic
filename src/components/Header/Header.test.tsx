import { shallow } from 'enzyme';
import React from 'react';
import { HeaderType } from '../../models/Types';
import Header from './Header';

describe( 'Header', () => {
    test( 'given size and value when render then show header', () => {
        const value = `test`;
        const headerSize = 1;
        const wrapper = shallow( <Header headerSize={ headerSize } value={ value }/> );
        expect( wrapper.exists( 'header' ) ).toBeTruthy();
        expect( wrapper.exists( 'h1' ) ).toBeTruthy();
        expect( wrapper.find( 'h1' ).text() ).toEqual( value );
    } );
    test( 'given className when render then show header with className', () => {
        const value = `test`;
        const headerSize = 5;
        const className: HeaderType = 'article-header';
        const wrapper = shallow( <Header headerSize={ headerSize } value={ value } className={ className }/> );
        expect( wrapper.is( '.article-header' ) ).toBeTruthy();
    } );
} );
