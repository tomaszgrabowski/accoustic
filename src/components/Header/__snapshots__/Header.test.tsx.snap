// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header given Header component when render then renders properly 1`] = `
<header>
  <h2>
    header
  </h2>
</header>
`;
